Utilities/docker

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+
+"""
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+ -------------------------------------------------------------------------
+ This script is used to bootstrap the Swift package manager build.
+
+ It does so by writing out a build task file which can be used to build a
+ stage1 package manager. That package manager is then expected to be able
+ to build itself to produce the production tools.
+
+ Note that currently this script is also responsible for building the package
+ manager in such a way that it can be installed along with the Swift package.
+ In particular, it knows how to build the runtime PackageDescription library
+ correctly and install it.  It can also build libSwiftPM, allowing clients to
+ access package manager functionality.
+"""
+
+import argparse
+import urllib2
+import os
+import subprocess
+import yaml
+
+def call(args):
+    """Prints and executes a command."""
+    print " ".join(args)
+    return subprocess.call(args)
+
+def docker_run(command_args):
+    """Runs a command in the container."""
+    call([
+        "docker",
+        "run",
+        "-it",
+        "--security-opt", "seccomp=unconfined",
+        "-v", os.getcwd() + ":/swiftpm",
+        "-w", "/swiftpm",
+        "--rm",
+        "swiftpm-docker-1604"
+    ] + command_args)
+
+def build(_):
+    """Builds a docker image with the latest snapshot."""
+
+    # Get latest snapshot info.
+    latest_build_url = "https://swift.org/builds/development/ubuntu1604/latest-build.yml"
+    latest_build_data = yaml.load(urllib2.urlopen(latest_build_url).read())
+    snapshot_filename = latest_build_data["download"]
+    # FIXME: We shouldn"t need to do this, it should be available in the API.
+    snapshot_name = snapshot_filename.replace("-ubuntu16.04.tar.gz", "")
+    base_url = latest_build_url.rsplit("/", 1)[0]
+    latest_snapshot_url = base_url + "/" + snapshot_name + "/" + snapshot_filename
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+
+    # Download latest snapshot (if necessary).
+    if snapshot_filename in os.listdir(script_dir):
+        print "Not downloading: " + latest_snapshot_url
```

Comment:
We should check the md5 here in case the current downloaded file is partial or incomplete.

Reply:
The `latest_build.yaml` file only provides the PGP signature. Is the md5 available somewhere?